### ISSUE TITLE
Pylance does not report multiple assignments to name in match pattern.

### ISSUE THREAD

**Issue by wanglc02** [microsoft/pyright#5742]
<h3> Environment data </h3>

-   Language Server version: 2023.8.30
-   OS and version: darwin arm64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: strict

<h3> Code Snippet </h3>

```python
class Point:
    def __init__(self, x: int, y: int) -> None:
        self.x = x
        self.y = y


match Point(3, 2):
    case Point(x=a, y=a):
        print(a)
```

<h3> Expected behavior </h3>

SyntaxError: multiple assignments to name 'a' in pattern

<h3> Actual behavior </h3>

Pylance does not report anything.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.324, which I just published. It will also be included in a future release of pylance.
